src/native/libs/Common/pal_config.h.in

```diff
@@ -8,13 +8,17 @@
 #cmakedefine01 HAVE_STATFS_MOUNT
 #cmakedefine01 HAVE_FLOCK64
 #cmakedefine01 HAVE_F_DUPFD_CLOEXEC
+#cmakedefine01 HAVE_F_DUPFD
```

Comment:
Do I understand correctly that this will be defined for all cases, including WASI? If so, how/when does the `!HAVE_F_DUPFD` code in `pal_io.c` get used?

If this is meant only for !wasi, then should it be in `#ifndef HOST_WASI` block at line 150?

Reply:
Is this being controlled by `check_symbol_exists`? If so, my change is likely incorrect?
```c
#ifndef HOST_WASI
#cmakedefine01 HAVE_GETRUSAGE
#endif
```
.. and should be changed to use the same `check_symbol_exists` mechanism?
But we also have this being set in `src/mono/cmake/configure.cmake`?

Reply:
libs and mono have different configure scripts (so, some of the things are twice or even one more for CLR)

Reply:
we shouldn't need to have the `#ifndef HOST_WASI` in this file since the configure script should set HAVE_GETRUSAGE based on whether the symbol is available.

Reply:
I removed `HAVE_GETRUSAGE` in favor of `-D_WASI_EMULATED_PROCESS_CLOCKS`